packages/app/src/cli/models/extensions/specification.ts

```diff
@@ -189,7 +189,7 @@ function appConfigTransform(
     const originPath = reverse ? mappedPath : objectPath
     const targetPath = reverse ? objectPath : mappedPath
     const sourceValue = getPathValue(content, originPath)
-    setPathValue(transformedContent, targetPath, sourceValue)
+    if (sourceValue !== undefined) setPathValue(transformedContent, targetPath, sourceValue)
```

Comment:
If the source is not found, shouldn't it fail? Or which scenario is this check covering?

Reply:
Already discussed and approved in the previous [PR](https://github.com/Shopify/cli/pull/3170)